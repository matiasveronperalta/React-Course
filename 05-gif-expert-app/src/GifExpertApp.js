import React, { useState } from 'react';


const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One punch', 'Dragon Ball', 'Naruto'])
    
    const handleAdd = () => {
        setCategories([...categories, 'Holi']);
    };

    return ( 
        <>
            <h2>GifExpertApp</h2>
            <hr />
            <button onClick={ handleAdd }> Agregar Categor√≠a </button>
            <ol>
                {
                    categories.map(category =>{
                        return <li key={category}> {category} </li>
                    })
                }
            </ol>
        </>
     );
}
 
export default GifExpertApp;