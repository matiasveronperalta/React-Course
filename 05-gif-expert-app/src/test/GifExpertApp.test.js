import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp.js'

describe('Test <GifExpertApp/>', () => {

    test('should be render and match with snapshoot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should be shoe category list', () => {
        
        const categories=['One Punch','Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})
