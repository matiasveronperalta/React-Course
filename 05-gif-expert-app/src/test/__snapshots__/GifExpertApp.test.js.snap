// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp/> should be render and match with snapshoot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test <GifExpertApp/> should be shoe category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
